pkg/cmd/kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,164 @@
+	cmd.Flags().StringVar(&o.Context, "context", o.Context, "The name of the kubeconfig context to use")
+
+	utilruntime.Must(cmd.RegisterFlagCompletionFunc("output", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return o.PrintFlags.AllowedFormats(), cobra.ShellCompDirectiveNoFileComp
+	}))
+
+	return cmd
+}
+
+// options is a struct to support kubeconfig command
+type options struct {
+	base.Options
+
+	// CurrentTarget is the current target
+	CurrentTarget target.Target
+
+	// PrintFlags composes common printer flag structs
+	PrintFlags *genericclioptions.PrintFlags
+	// ResourcePrinterFunc is a function that can print objects
+	PrintObject printers.ResourcePrinterFunc
+
+	// Flatten flag the resulting kubeconfig file into self-contained output
+	Flatten bool
+	// Minify flag to remove all information not used by current-context from the output
+	Minify bool
+	// RawByteData flag to display raw byte data
+	RawByteData bool
+
+	// Context holds the name of the kubeconfig context to use
+	Context string
+}
+
+// newOptions returns initialized options
+func newOptions(ioStreams util.IOStreams) *options {
+	return &options{
+		Options: base.Options{
+			IOStreams: ioStreams,
+		},
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme).WithDefaultOutput("yaml"),
+	}
+}
+
+// Complete adapts from the command line args to the data required.
+func (o *options) Complete(f util.Factory, _ *cobra.Command, _ []string) error {
+	manager, err := f.Manager()
+	if err != nil {
+		return err
+	}
+
+	o.CurrentTarget, err = manager.CurrentTarget()
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObject = printer.PrintObj
+
+	return nil
+}
+
+// Validate validates the provided command options.
+func (o *options) Validate() error {
+	if o.CurrentTarget.GardenName() == "" {
+		return target.ErrNoGardenTargeted
+	}
+
+	return nil
+}
+
+// Run does the actual work of the command.
+func (o *options) Run(f util.Factory) error {
+	ctx := f.Context()
+
+	manager, err := f.Manager()
+	if err != nil {
+		return err
+	}
+
+	config, err := manager.ClientConfig(ctx, o.CurrentTarget)
+	if err != nil {
+		return err
+	}
+
+	rawConfig, err := config.RawConfig()
+	if err != nil {
+		return err
+	}
+
+	if o.Minify {
+		if len(o.Context) > 0 {
+			rawConfig.CurrentContext = o.Context
+		}
+
+		if err := clientcmdapi.MinifyConfig(&rawConfig); err != nil {
+			return err
+		}
+	}
+
+	if o.Flatten {
+		if err := clientcmdapi.FlattenConfig(&rawConfig); err != nil {
+			return err
+		}
+	} else if !o.RawByteData {
+		clientcmdapi.ShortenConfig(&rawConfig)
+	}
+
+	convertedObj, err := clientcmdlatest.Scheme.ConvertToVersion(&rawConfig, clientcmdlatest.ExternalVersion)
+	if err != nil {
+		return err
+	}
+
+	return o.PrintObject(convertedObj, o.IOStreams.Out)
```

Comment:
Add TODO that we will align PrintObject in base and this impl.

Reply:
✅